packages/slate/src/create-editor.ts

```diff
@@ -57,13 +57,17 @@ export const createEditor = (): Editor => {
       }
 
       const oldDirtyPaths = DIRTY_PATHS.get(editor) || []
-      const newDirtyPaths = getDirtyPaths(op)
-
+      const canTransformPath = Path.operationCanTransformPath(op)
       for (const path of oldDirtyPaths) {
-        const newPath = Path.transform(path, op)
-        add(newPath)
+        if (canTransformPath) {
```

Comment:
I was hoping we could optimize this more but I realized we have to rebuild the set for de-duping each time. Maybe the key set should be persisted with the DIRTY_PATHS so this doesn't need to be reconstructed each time. This code path can be so hot on large edits.

Reply:
I can investigate storing the path keys alongside the dirty paths and seeing what additional savings we can get

Reply:
Updated to reuse the array/set entirely if the op won't touch them. Shaved an additional ~10% of time off the test mentioned in the summary